ci/scripts/integration-test.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+require() {
+	if [ "$1" == "" ]
+	then
+		echo "The environment variable $2 must be present and contain $3"
+		exit 4
+	fi
+}
+
+section() {
+	echo -e "\n============================\n= $1\n============================\n"
+}
+
+SCRIPTDIR=$(dirname $0)
+SCRIPTDIR=$(realpath $SCRIPTDIR)
+
+section "Checking config"
+require "$SERVICE_ACCOUNT" "SERVICE_ACCOUNT" "the service account JSON pubsubbeat will use"
+require "$PROJECT_ID" "PROJECT_ID" "the GCP project ID"
+require "$TOPIC_NAME" "TOPIC_NAME" "the name of the integration test pubsub topic"
+require "$SUBSCRIPTION_NAME" "SUBSCRIPTION_NAME" "the name of the integration test subscription"
+
+section "Building"
+source $SCRIPTDIR/setup-go.sh
+make
+
+section "Pulling down gcloud"
+source $SCRIPTDIR/setup-gcloud.sh
+
+section "Setting up environment"
+export TESTID=`date +%s`
+export TESTDIR="/tmp/$TESTID"
+export KEYPATH="$TESTDIR/key.json"
+
+mkdir -p $TESTDIR
+
+echo "Test ID: $TESTID"
+echo $SERVICE_ACCOUNT > $KEYPATH
+
+section "Clearing any existing messages from $SUBSCRIPTION_NAME"
+gcloud pubsub subscriptions pull --limit 100 --auto-ack "projects/$PROJECT_ID/subscriptions/$SUBSCRIPTION_NAME"
```

Comment:
Unless a limit of 100 is intentional, consider using "unlimited" which ensures full clearance in the case of 100+ messages, see: https://cloud.google.com/sdk/gcloud/reference/alpha/pubsub/subscriptions/pull

Reply:
Disregard. I tried --limit unlimited which doesn't seem to actually work.